Dockerfile

```diff
@@ -6,61 +6,35 @@ ARG GIT_BRANCH
 # Adding rust binaries to PATH.
 ENV PATH="$PATH:/root/.cargo/bin"
 
-RUN apt-get update
-
-RUN apt-get -y install gcc
-
-# Installing rustup.
-RUN apt-get -y install curl
-# Install a fixed version of rust.
+# Install all required packages in one go to optimize the image
+# https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#run
+# DEBIAN_FRONTEND is set for tzdata.
+RUN apt-get update && \
+    DEBIAN_FRONTEND="noninteractive" apt-get install --no-install-recommends -y \
+    curl gcc git python3 python3-pip shellcheck \
```

Comment:
Will this always install the latest python or something like that (I noticed the 3.6 -> 3.8 update in docs below)? Should we stick to a fixed version?